language/move-vm/runtime/src/interpreter.rs

```diff
@@ -58,16 +58,16 @@ macro_rules! set_err_info {
 ///
 /// An `Interpreter` instance is a stand alone execution context for a function.
 /// It mimics execution on a single thread, with an call stack and an operand stack.
-pub(crate) struct Interpreter<L: LogContext> {
+pub(crate) struct Interpreter<'a> {
     /// Operand stack, where Move `Value`s are stored for stack operations.
     operand_stack: Stack,
     /// The stack of active functions.
     call_stack: CallStack,
     // Logger to report information to clients
-    log_context: L,
+    log_context: &'a dyn LogContext,
```

Comment:
Yeah this one is a little bit funky. I'll definitely look into logging in a more holistic way later, for this and another issue (Move VM using Diem logging traits etc.)

> It took a potentially owned version before
Yes and no. Previously the entry point took a reference to a context and clone it. It feels odd to me that the VM is making the assumption that the context can be cloned so I took the safe route of not assuming so. A box would also work and it would be more similar to what it is before.

> It feels strange to me that the invoker of the VM would need the particular instance of the log context back
This I'm not so sure. Intuitively it's not so absurd to assume that a system-wide log environment would outlive this particular Move VM instance and continue to serve other activities. I could be wrong due to my limited familiarity with the diem logging framework. As I mentioned above, some further investigation is needed here.

Reply:
A box here just feels simpler, and is less code movement than what was previously there in terms of ownership. So I would vote to just do that

Reply:
FYI I'll just leave this as is since we're planning to replace `diem-logger` with `tracing` and once that's done the log_context will no longer be necessary.